geometry/rp2_element.hpp

```diff
@@ -0,0 +1,65 @@
+﻿
+#pragma once
+
+#include <string>
+
+#include "geometry/r3_element.hpp"
+
+namespace principia {
+namespace geometry {
+namespace internal_rp2_element {
+
+// An |RP2Element| is an element of ℝP², the real projective plane.  |Scalar|
+// must be a 1-dimensional vector space over ℝ, typically represented by
+// |Quantity| or |double|.
+template<typename Scalar>
+class RP2Element {
+ public:
+  // Constructs a point from a set of homogeneous coordinates.  The
+  // coordinates may not all be zero.
+  explicit RP2Element(Scalar const& x, Scalar const& y, double z);
+
+  // Returns true if and only if the point is at infinity.
+  bool is_at_infinity() const;
+
+  // Returns the Euclidean (inhomogeneous) coordinates of the point.  May be
+  // infinities.
```

Comment:
Document the semantics: the signs of (x,y) will be in the quadrant determined by the signs of x and y, and both will be infinite or none.